pkg/controllers/jobset_controller.go

```diff
@@ -218,6 +224,69 @@ func (r *JobSetReconciler) getChildJobs(ctx context.Context, js *jobset.JobSet)
 	}
 	return &ownedJobs, nil
 }
+func (r *JobSetReconciler) calculateAndUpdateReplicatedJobsStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs) error {
+	status, err := r.calculateReplicatedJobStatuses(ctx, js, jobs)
+	if err != nil {
+		return err
+	}
+	// Check if status ReplicatedJobsStatus has changed
+	if apiequality.Semantic.DeepEqual(js.Status.ReplicatedJobsStatus, status) {
+		return nil
+	}
+	js.Status.ReplicatedJobsStatus = status
+	return r.Status().Update(ctx, js)
+}
+
+func (r *JobSetReconciler) calculateReplicatedJobStatuses(ctx context.Context, js *jobset.JobSet, jobs *childJobs) ([]jobset.ReplicatedJobStatus, error) {
+	log := ctrl.LoggerFrom(ctx)
+
+	// Prepare replicatedJobsReady for optimal iteration
+	replicatedJobsReady := map[string]int32{}
+	for _, replicatedJob := range js.Spec.ReplicatedJobs {
+		replicatedJobsReady[replicatedJob.Name] = 0
+	}
+
+	// Calculate jobsReady for each Replicated Job
+	for _, job := range append(jobs.active, append(jobs.successful, jobs.failed...)...) {
+		ready := pointer.Int32Deref(job.Status.Ready, 0)
+		// parallelism is always set as it is otherwise defaulted by k8s to 1
+		podsCount := *(job.Spec.Parallelism)
+		if job.Spec.Completions != nil && *job.Spec.Completions < podsCount {
+			podsCount = *job.Spec.Completions
+		}
+		if job.Status.Succeeded+ready >= podsCount {
+			if job.Labels != nil && job.Labels[jobset.ReplicatedJobNameKey] != "" {
+				replicatedJobsReady[job.Labels[jobset.ReplicatedJobNameKey]]++
+			} else {
+				log.Error(nil, fmt.Sprintf("job %s missing ReplicatedJobName label", job.Name))
+			}
+		}
+	}
+
+	// Update current ReplicatedJobsStatuses
+	rjStatus := js.Status.DeepCopy().ReplicatedJobsStatus
+	for index := range rjStatus {
+		val, ok := replicatedJobsReady[rjStatus[index].Name]
+		if !ok {
+			// the current replicated job is not anymore relevant
+			rjStatus = append(rjStatus[:index], rjStatus[index+1:]...)
+		} else {
+			rjStatus[index].ReadyJobs = val
+		}
+		// Delete updated ReplicatedJobsReady from replicatedJobsReady
+		// it will be used to append non-existent ones
+		delete(replicatedJobsReady, rjStatus[index].Name)
+	}
```

Comment:
The set of replicatedJobs is immutable, so this loop is not necessary, just calculate ReplicatedJobsStatus from scratch below like this:

``
rjStatus := []ReplicatedJobStatus{}
...
return rjStatus
```